Sources/MobileSdk/StorageManager.swift

```diff
@@ -16,129 +32,157 @@ import Foundation
+
+            let asdir = try FileManager.default.url(for: .applicationSupportDirectory,
+                                                    in: .userDomainMask,
+                                                    appropriateFor: nil, // Ignored
+                                                    create: true) // May not exist, make if necessary.
+
+            return asdir.appendingPathComponent(file)
+        } catch { // Did the attempt to get the application support dir fail?
+            print("Failed to get/create the application support dir.")
+            return nil
+        }
+    }
+
+    // Method: add()
+    //    Store a value for a specified key, encrypted in place.
+    //
+    // Arguments:
+    //     key   - the name of the file
+    //     value - the data to store
+    //
+    // Returns:
+    //    A boolean indicating success.
+
+    func add(key: Key, value: Value) throws {
+        guard let file = path(file: key) else { return }
+
+        do {
+            try value.write(to: file, options: .completeFileProtection)
+        } catch {
+            print("Failed to write the data for '\(key)'.")
+            return
+        }
+    }
+
+    // Method: get()
+    //    Get a value for the specified key.
+    //
+    // Arguments:
+    //    key - the name associated with the data
+    //
+    // Returns:
+    //    Optional data potentially containing the value associated with the key; may be `nil`.
+
+    func get(key: Key) throws -> Value {
+        guard let file = path(file: key) else { return Data() }
+
+        do {
+            let d = try Data(contentsOf: file)
+            return d
+        } catch {
+            print("Failed to read '\(file)'.")
+        }
+
+        return Data()
+    }
+
+    // Method: list()
+    //    List the the items in storage.  Note that this will list all items in the `application support` directory,
+    // potentially including any files created by other systems.
+    //
+    // Returns:
+    //    A list of items in storage.
+
+    func list() -> [Key] {
+        guard let asdir = path(file: "")?.path else { return [String]() }
+
+        do {
+            return try FileManager.default.contentsOfDirectory(atPath: asdir)
+        } catch {
+            return [String]()
+        }
+    }
+
+    // Method: remove()
+    //    Remove a key/value pair. Removing a nonexistent key/value pair is not an error.
+    //
+    // Arguments:
+    //    key - the name of the file
+    //
+    // Returns:
+    //    A boolean indicating success; at present, there is no failure path, but this may change in the future.
+
+    func remove(key: Key) throws {
+        guard let file = path(file: key) else { return }
+
+        do {
+            try FileManager.default.removeItem(at: file)
+        } catch {
+            // It's fine if the file isn't there.
+        }
+    }
+
+    // Method: sys_test()
+    //    Check to see if everything works.
+
+    /*
+     func sys_test() {
```

Comment:
Can you set this up as a unit test that runs on the CI?

Reply:
Done.

Reply:
Could you remove this comment now that it's setup as a unit test?